### PR TITLE
Alfredapi 422 fix totalItemsReturned for TDMQ's

### PR DESCRIPTION
**Pull request by RVanhuysseXenit**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-422

- [X] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [X] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [X] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation)?
- [X] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [X] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

File: apix-impl/build.gradle [hunk 1]
```diff
@@ -58,7 +58,7 @@ subprojects {
         compile group: 'javax.validation', name: 'validation-api', version: '1.1.0.Final'
 
         testCompile group: 'eu.xenit.de', name: 'annotations', version: de_version
-        testCompile group: 'org.alfresco', name: 'alfresco-repository', version: alfresco_version
+        testCompile group: 'org.alfresco', name: 'alfresco-repository', version: alfresco_repo_version , classifier: 'sources'
```
Comment (thijslemmens) [inline apix-impl/build.gradle:61]: why sources?

Comment (RVanhuysseXenit) [reply apix-impl/build.gradle:61]: hmm, seems I've got some cleanup to do.

**Comment by WimCrols** [comment]
All commits require ticket ID.

File: apix-docker/build.gradle [hunk 2]
```diff
@@ -26,7 +26,7 @@ subprojects {
     dockerCompose {
         useComposeFiles = ['docker-compose.yml']
         // Don't use the dev compose file on Jenkins
-        if(ci.isCi()) {
```
Comment (WimCrols) [inline apix-docker/build.gradle:29]: Woops... that slipped through

**Comment by RVanhuysseXenit** [comment]
@WimCrols will be cleaned up with rebase

File: apix-impl/src/integration-test/java/eu/xenit/apix/tests/search/SearchServiceTest.java [hunk 3]
```diff
@@ -154,18 +157,45 @@ public void TestSkip() throws IOException, InterruptedException {
 
     @Test
     public void TestTotalCount() throws IOException, InterruptedException {
+        transactionService.getRetryingTransactionHelper()
```
Comment (WimCrols) [inline apix-impl/src/integration-test/java/eu/xenit/apix/tests/search/SearchServiceTest.java:160]: Too bad we need to put sleeps here... Please add to scope of https://xenitsupport.jira.com/browse/ALFREDAPI-425 the location we need to clean up. So in this case, this sleep etc.

File: apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchService.java [hunk 4]
```diff
@@ -150,12 +150,12 @@ protected SearchParameters buildSearchParameters(SearchQuery postQuery) {
 
     // These 2 methods are overridden in the 4.2 impl
     protected void setSearchLimit(SearchParameters searchParameters, int max) {
-        searchParameters.setMaxItems(max);
+        searchParameters.setMaxItems(-1);
```
Comment (WimCrols) [inline apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchService.java:153]: This magic number can use a comment :)

**Pull request merged**